Sources/Tracking/Service/HttpRequestRunner.swift

```diff
@@ -57,6 +58,33 @@ internal class UrlRequestHttpRequestRunner: HttpRequestRunner {
         }.resume()
     }
 
+    public func downloadFile(url: URL, onComplete: @escaping (URL?) -> Void) {
+        guard let documentsDirectoryURL = FileManager.default.urls(for: .cachesDirectory, in: .userDomainMask).first
+        else { return }
+
+        session.downloadTask(with: url) { tempLocation, _, _ in
+            guard let tempLocation = tempLocation else {
+                return onComplete(nil)
+            }
+
+            // note make sure that the file name below is unique to not overwrite previous file.
+            var uniqueFileName = String.random
+            if let fileExtension = url.absoluteString.fileExtension {
+                uniqueFileName += ".\(fileExtension)"
+            }
+
+            let destinationURL = documentsDirectoryURL.appendingPathComponent(uniqueFileName)
```

Comment:
is there any value in putting these in a `cio` subdirectory in the cache or something?

Reply:
additionally, in a bunch of our Go code we use a `LimitReader` to avoid loading an unbounded-size file, does it make sense to do that in the SDK as well?

Reply:
> is there any value in putting these in a cio subdirectory in the cache or something?
Yes, I believe so. Good idea. 

> additionally, in a bunch of our Go code we use a LimitReader to avoid loading an unbounded-size file, does it make sense to do that in the SDK as well?
I created [a ticket](https://github.com/customerio/customerio-ios/issues/61) for this. I suggest we do this another day because (1) it will take a bit more work to get this working and (2) we talked about for `alpha` we suggest to customers in the docs to limit their file size of the image they want to show.